MarkdownLivePreview.py

```diff
@@ -169,72 +218,76 @@ def on_close(self, markdown_view):
-        sublime.set_timeout(partial(self._update_preview, markdown_view), DELAY)
+        sublime.set_timeout(partial(self._update_preview, view), DELAY)
 
-    def _update_preview(self, markdown_view):
-        # if the buffer id is 0, that means that the markdown_view has been closed
-        # This check is needed since a this function is used as a callback for when images
-        # are loaded from the internet (ie. it could finish loading *after* the user
-        # closes the markdown_view)
+    def _update_preview(self, view):
+        # if the buffer id is 0, that means that the view has been
+        # closed This check is needed since a this function is used as a
+        # callback for when images are loaded from the internet (ie. it could
+        # finish loading *after* the user closes the view)
         if time.time() - self.last_update < DELAY / 1000:
             return
 
-        if markdown_view.buffer_id() == 0:
+        # Desc: Check if View is in Previews
+        if view.id() not in PREVIEW_VIEWS.keys():
             return
 
-        self.last_update = time.time()
+        preview_view = None
+        for enum_view in sublime.active_window().sheets():
+            if enum_view.id() == PREVIEW_VIEWS[view.id()]:
+                preview_view = enum_view
+                break
 
-        total_region = sublime.Region(0, markdown_view.size())
-        markdown = markdown_view.substr(total_region)
+        # Desc: If Preview is None, we can't update
+        if preview_view is None:
+            return
 
-        preview_view = markdown_view.window().active_view_in_group(1)
-        viewport_width = preview_view.viewport_extent()[0]
+        if view.buffer_id() == 0:
+            return
 
-        basepath = os.path.dirname(markdown_view.file_name())
-        html = markdown2html(
-            markdown,
-            basepath,
-            partial(self._update_preview, markdown_view),
-            resources,
-            viewport_width,
+        self.last_update = time.time()
+
+        total_region = sublime.Region(0, view.size())
+        content = "{}\n\n{}".format(
+            mdpopups.format_frontmatter(frontmatter),
+            self.render_checkboxes(view.substr(total_region)),
         )
+        html_content = mdpopups.md2html(view, content).replace("<br>", "<br/>")
 
-        self.phantom_sets[markdown_view.id()].update(
-            [
-                sublime.Phantom(
-                    sublime.Region(0),
-                    html,
-                    sublime.LAYOUT_BLOCK,
-                    lambda href: sublime.run_command("open_url", {"url": href}),
-                )
-            ]
+        file_name = view.file_name()
+        basepath = os.path.dirname(file_name) if file_name else None
+        html_content = imageparser(
+            html_content, basepath, partial(self._update_preview, view), resources,
         )
 
+        mdpopups.update_html_sheet(preview_view, html_content, md=False)
```

Comment:
I suggest changing this to:

`mdpopups.update_html_sheet(sheet=preview_view, contents=html_content, md=False)`